arkouda/dataframe.py

```diff
@@ -2102,6 +2102,101 @@ def unregister_dataframe_by_name(user_defined_name: str) -> None:
         unregister_pdarray_by_name(f"df_index_{user_defined_name}_key")
         Strings.unregister_strings_by_name(f"df_columns_{user_defined_name}")
 
+    @staticmethod
+    def _parse_col_name(entryName, dfName):
+        """
+        Helper method used by from_return_msg to parse the registered name of the data component
+        and pull out the column type and column name
+
+        Parameters
+        ----------
+        entryName : string
+            The full registered name of the data component
+
+        dfName : string
+            The name of the DataFrame
+
+        Returns
+        -------
+        Tuple (columnName, columnType)
+        """
+        regName = entryName.split(" ")[1]
+        colParts = regName.split("_")
+        colType = colParts[2]
+
+        # Case of '_' in the column or dataframe name
+        if len(colParts) > 5:
+            nameInd = regName.rindex(dfName) - 1
+            startInd = len(colType) + 9
+            return regName[startInd:nameInd], colType
+        else:
+            return colParts[3], colType
+
+    @staticmethod
+    def from_return_msg(repMsg):
+        """
+        Creates and returns a DataFrame based on return components from ak.util.attach
+
+        Parameters
+        ----------
+        repMsg : string
+            A '+' delimited string of the DataFrame components to parse.
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame representing a set of DataFrame components on the server
+
+        Raises
+        ------
+        RuntimeError
+            Raised if a server-side error is thrown in the process of creating
+            the DataFrame instance
+        """
+        parts = repMsg.split("+")
```

Comment:
Is there any way we could make the return from the a JSON format here? Everything you have looks to check out, but it seems like reading things out of a keyed dictionary would make it much more simple and easier to read.

Reply:
Possible - Yes. 

However that would require changing how all of the types handle the return and not just DataFrame (Since DataFrame calls all of the other types using the repMsg parts) so I think this should be a separate issue

Reply:
10-4, let's just put in a separate issue for that then and work it when time permits.